### ISSUE TITLE
will flickity support large galleries (unlike current isotope.js)?

### ISSUE THREAD

**Issue by ngault** [metafizzy/flickity#2]
Flickity looks exciting, especially to current Isotope and Masonry users.  

However, as somebody that can't move to Isotope v2 because of its inability to handle > 100  (see issues #718,#677), I wonder whether Flickity will also be limited to handling small galleries?   Flickity looks to have the same source dependencies as Isotope...

I sympathize you've been grappling with this annoying problem for a while now with Isotope.js, but, at least for some of us, it does cast a cloud over Flickity.

**Comment by desandro** [comment]
Possibly.

But like Isotope and Masonry, Flickity will not be optimized for hundreds of cells. I feel that's an edge case that is not a prioritizing, at least for now. My hunch is that most use cases will be less than 100 cells.

I should clarify that Flickity, which handles galleries/sliders/carousels, is significantly different from Isotope and Masonry, which handles grid layouts. It's two different use cases.

**Comment by ngault** [comment]
Dave, thanks for the quick response.  I respect that you have to prioritize the most important uses of your products, and that my interests may not be representative.  Still, as a heavily invested user in your projects, I offer two ideas for your consideration in planning future development:

1) Although Isotope and Masonry may typically have a different use case (namely, grids) than the one you contemplate for Flickity (sliders/galleries), taken together they may represent a complete solution for front-ends across desktop desktops to tablets to mobile phones.  Isotope v1 has proven a great solution for grid-based exploration of large datasets on the desktop and tablets, but I've found a 2D grid works less well on a mobile phone.  Flickity, with its single row of items and support for gestures/touch events, looked to me like the perfect mobile phone front-end for the same data-intensive back-end that Isotope currently serves so well.  (I'm currently using a lot of mobile-specific hacks to make Isotope work better on mobile phones.  Not ideal.)  With the addition of Flickity to your other projects, you are in the perfect position to offer a complete solution across devices.  Each project can be positioned both as stand-alone and as part of an integrated, complete offering. As an example, in my case, a pre-existing Isotope implementation enables an almost instant implementation (and purchase!) of Flickity for display of the same data on phones.

2) With so many users trying to figure out how Isotope, Masonry, and (soon) Flickity fit with JavaScript frameworks (like React and Angular) and Web Components, now is the time to provide a solution that would not only put all this confusion to rest, but also significantly increase their use.  The use of grids and galleries with JavaScript frameworks (as opposed to standard web pages) is defined by access to larger datasets and virtualized DOMs.  You may take the position that this area is still too unsettled (virtual DOMs, shadow DOM's, dirty checking, etc, etc), but I believe that the right design for your projects is clear.  In short, as all advanced Web components will, you should be managing/virtualizing your DOM, enabling support for unlimited numbers of items across all your projects.  I highly recommend this article as suggesting the way forward:
http://futurice.com/blog/combining-react-flux-and-web-components

That's my perspective, FWIW.  I'd love to continue using your projects and not be forced to adopt lesser alternatives that offer only the advantage of working with networked data.

Thanks!

**Comment by desandro** [comment]
Thank you for these insights and your astute analysis. Your thoughtfulness helps me gain better perspective.

**tldr** For the initial release, I'd like to execute on the current feature set. I like both of your ideas and I may pursue them in the future.

### 1. Combining Flickity and Isotope

True, the opportunity may be there. But I already feel that Isotope is cumbersome, and Flickity is growing cumbersome as well in its own feature set. I see combining them could lead to a quagmire. I am wary of building one layout library to rule them all. You're the first to bring this up, so I'll consider it more as others voice their consent.

### 2. Supporting frameworks like Angular and React

Integration with these frameworks has been something I've been tiptoeing around. My feeling is that, while these frameworks sound incredibly popular at the moment, there is still a larger demand for simpler `<script />` usage, like in Wordpress themes, and smaller sites. The frameworks are likely to be the future, but they are too cutting edge to commit to right now (future seems uncertain for Angular with the v2.0 upheaval). If I were to support these frameworks, I want to be best-in-class at it, which means getting familiar with them. I am no where near that now (See metafizzy/isotope#605), so that would be just as a big of an undertaking as adding the support to the Metafizzy libraries.

I know these frameworks are the future, and Metafizzy libraries could benefit greatly by properly supporting them. But for now, I'd like to focus on executing on the original feature set before considering this huge task.

**Comment by juliocesar** [comment]
@desandro Rest easy. The API for your libraries takes an element. Integration on both counts is *very* trivial. :tada:

**Comment by desandro** [comment]
Closing this issue for now. Using Flickity with React and Angular are still pertinent issues, but this discussion in particular has cooled off :shaved_ice:

**Comment by ngault** [comment]
Hi David,

Thanks for the heads-up that Flickity won’t be supporting frameworks anytime soon, and I understand your decision.  But, to be clear, that discussion was incidental to the open issue, which was:
...as somebody that can't move to Isotope v2 because of its inability to handle > 100 (see issues #718,#677), I wonder whether Flickity will also be limited to handling small galleries? Flickity looks to have the same source dependencies as Isotope...

I sympathize you've been grappling with this annoying problem for a while now with Isotope.js, but, at least for some of us, it does cast a cloud over Flickity.

With respect to Flickity, this limitation can’t fairly be called a bug because the product has never shipped.  But if you were a paying customer for Isotope.js, consider how you’d feel if you couldn’t update from a very old version used by your existing enterprise app because the new versions of Isotope (which fix other reported bugs) completely break the user experience.  That’s the situation I’m in.

Best,
Nick Gault

**Comment by robrecord** [comment]
"You're the first to bring this up, so I'll consider it more as others voice their consent."

Please add my voice. I am currently attempting to combine Flickity and Isotope for the same content, for the purpose that @ngaul outlines - best user experience across desktop and mobile form factors.

**Comment by reberhardt** [comment]
re: combining flickity and isotope, 

I just did something like this for http://chicosfas-cares.com/kids, the isotope/flickity combo was a perfect solution for presenting the portraits in a responsive way

would be great to see some integration between the two products down the road
